File: languages/php/Bloock/Entity/Identity/StringAttribute.php
<?php

namespace Bloock\Entity\Identity;

/**
 * Represents an attribute with a string value.
 */
class StringAttribute extends Attribute
{
    public static function fromProto(\Bloock\StringAttribute $res): StringAttribute
    {
        return new StringAttribute($res->getId(), $res->getValue());
    }

    public function toProto(): \Bloock\StringAttribute
    {
        $p = new \Bloock\StringAttribute();
        $p->setId($this->id);
        $p->setValue($this->value);
        return $p;
    }
}

File: languages/python/bloock/entity/identity/string_attribute.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_pb2 as proto
from bloock.entity.identity.attribute import Attribute


class StringAttribute(Attribute):
    """
    Represents an attribute with a string value.
    """
    @staticmethod
    def from_proto(a: proto.StringAttribute) -> StringAttribute:
        return StringAttribute(
            id=a.id,
            value=a.value,
        )

    def to_proto(self) -> proto.StringAttribute:
        return proto.StringAttribute(
            id=self.id,
            value=self.value,
        )
